prow/gerrit/adapter/adapter.go

```diff
@@ -692,6 +697,30 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 	return nil
 }
 
+// lastRevisionWithCodeChange tries to find a revision with code change since last update.
+// There is no point repeating tests if no code change happened since last update.
+func lastRevisionWithCodeChange(change client.ChangeInfo, lastUpdate time.Time) (client.RevisionInfo, error) {
+	var revisionsWithCodeChange []client.RevisionInfo
+	for _, revision := range change.Revisions {
+		if revision.Kind == gerrit.NoCodeChange {
+			continue
+		}
+		revisionsWithCodeChange = append(revisionsWithCodeChange, revision)
+	}
+	// sorts in reverse chronological order
+	sort.Slice(revisionsWithCodeChange, func(i, j int) bool {
+		return revisionsWithCodeChange[i].Created.Time.After(revisionsWithCodeChange[j].Created.Time)
+	})
+
+	for _, revision := range revisionsWithCodeChange {
+		if revision.Created.Time.After(lastUpdate) {
+			return revision, nil
+		}
+	}
+
+	return client.RevisionInfo{}, fmt.Errorf("cannot find a new revision with code change after last update for repo %s, change %s", change.Project, change.ChangeID)
```

Comment:
This shouldn't result in us logging an error (line 601) since this is a normal outcome, not an error, and we don't want to spam our logs.
It doesn't look like this function can ever encounter an error so I think it would make more sense for it to return either `(client.RevisionInfo, bool)` or simply `*client.RevisionInfo` to optionally return a revision.

Reply:
Good point, done.